packages/utils/package.json

```
@@ -0,0 +1,51 @@
+{
+	"name": "@directus/utils",
+	"version": "9.21.2",
+	"type": "module",
+	"scripts": {
+		"build": "tsc --build",
+		"dev": "tsc --watch",
+		"test": "vitest run",
+		"test:watch": "vitest",
+		"test:coverage": "vitest run --coverage"
+	},
+	"description": "Utilities shared between the Directus packages",
+	"repository": {
+		"type": "git",
+		"url": "https://github.com/directus/directus.git",
+		"directory": "packages/utils"
+	},
+	"funding": "https://github.com/directus/directus?sponsor=1",
+	"license": "GPL-3.0",
+	"author": "Rijk van Zanten <rijkvanzanten@me.com>",
+	"exports": {
+		".": {
+			"import": "./dist/index.js",
+			"types": "./dist/index.d.ts"
+		},
+		"./node": {
+			"import": "./dist/node/index.js",
+			"types": "./dist/node/index.d.ts"
+		},
+		"./package.json": "./package.json"
+	},
```

According to the [TS docs](https://www.typescriptlang.org/docs/handbook/esm-node.html#packagejson-exports-imports-and-self-referencing) the "types" property must occur first:
```jsonc
    "exports": {
        ".": {
            // Entry-point for TypeScript resolution - must occur first!
            "types": "./types/index.d.ts",
            // Entry-point for `import "my-package"` in ESM
            "import": "./esm/index.js",
            // Entry-point for `require("my-package") in CJS
            "require": "./commonjs/index.cjs",
        },
    },
```

Not sure if this only applies when having both "import" and "require", but better be safe:

```suggestion
	"exports": {
		".": {
			"types": "./dist/index.d.ts",
			"import": "./dist/index.js"
		},
		"./node": {
			"types": "./dist/node/index.d.ts",
			"import": "./dist/node/index.js"
		},
		"./package.json": "./package.json"
	},
```

---

TypeScript never seizes to amaze me 🥲 I guess it makes sense though! It'll go top to bottom figuring out which thing to use, and short-circuits on the first hit. Good catch!

---

> TypeScript never seizes to amaze me 🥲

You have taken the words out of my mouth 🤣😢

> I guess it makes sense though! It'll go top to bottom figuring out which thing to use, and short-circuits on the first hit. Good catch!

Yes, I think so too.